### ISSUE TITLE
Beit why no k bias in attention?

### ISSUE THREAD

**Issue by cqliheping** [microsoft/unilm#510]
The k bias is always zero in code. Is there any reason for this?  This is different from the normal implement.
 https://github.com/microsoft/unilm/blob/421cffe163e474189c410ee06dad48dbfdcbc135/beit/modeling_finetune.py#L124

In my test.  when finetune, k bias has little affect on performance.  But I do not have a test on pretrain.

**Comment by donglixp** [comment]
Both (i.e., with or without key.bias) are equivalent in terms of calculation results. They are canceled by the softmax function.
 
Softmax(q,k) = exp(q.weight * key.weight + q.bias * key.weight + q.weight * key.bias + q.bias * key.bias) / Z
 
Because the query is the same over all the keys, so the term (q.weight * key.bias + q.bias * key.bias) remains the same across all the keys, which in turn can be cancelled without affecting the softmax results.
 
exp(a)/(exp(a)+ exp(b)) == exp(a+C)/(exp(a+C)+ exp(b+C))
